stan/math/rev/core/operator_division.hpp

```diff
@@ -126,7 +104,189 @@ inline var operator/(const var& dividend, Arith divisor) {
+ *
+ * @tparam Mat Either a type inheriting from `EigenBase` with a scalar type of
+ * `var` or a `var_value<T>` with type `T` inheriting from `EigenBase`.
+ * @param[in] m specified matrix or expression
+ * @param[in] c specified scalar
+ * @return matrix divided by the scalar
+ */
+template <typename Mat, require_matrix_st<is_var, Mat>* = nullptr>
+inline auto divide(var c, const Mat& m) {
+  arena_t<plain_type_t<Mat>> arena_m = m;
+  auto inv_m = to_arena(1.0 / arena_m.val().array());
+  arena_t<plain_type_t<Mat>> res = c.val() * inv_m;
+  reverse_pass_callback([c, inv_m, arena_m, res]() mutable {
+    arena_m.adj().array() -= inv_m * res.adj().array() * res.val().array();
+    c.adj() += (inv_m * res.adj().array()).sum();
+  });
+  return plain_type_t<Mat>(res);
+}
+
+/**
+ * Return scalar divided by matrix.
+ *
+ * @tparam Mat A type inheriting from `EigenBase` with an `Arithmetic` scalar
+ * type.
+ * @param[in] c specified scalar
+ * @param[in] m specified matrix or expression
+ * @return matrix divided by the scalar
+ */
+template <typename Mat, require_eigen_vt<std::is_arithmetic, Mat>* = nullptr>
+inline auto divide(var c, const Mat& m) {
+  arena_t<Mat> arena_m = m;
+  auto inv_m = to_arena(1.0 / arena_m.val().array());
+  arena_t<promote_scalar_t<var, Mat>> res = c.val() * inv_m;
+  reverse_pass_callback([c, inv_m, res]() mutable {
+    c.adj() += (inv_m * res.adj().array()).sum();
+  });
+  return promote_scalar_t<var, Mat>(res);
+}
+
+/**
+ * Return scalar divided by matrix.
+ *
+ * @tparam Mat Either a type inheriting from `EigenBase` with a scalar type of
+ * `var` or a `var_value<T>` with type `T` inheriting from `EigenBase`.
+ * @param[in] c specified scalar
+ * @param[in] m specified matrix or expression
+ * @return matrix divided by the scalar
+ */
+template <typename Mat, require_matrix_st<is_var, Mat>* = nullptr>
+inline auto divide(double c, const Mat& m) {
+  arena_t<promote_scalar_t<var, Mat>> arena_m = m;
+  auto inv_m = to_arena(1.0 / arena_m.val().array());
+  arena_t<promote_scalar_t<var, Mat>> res = c * inv_m;
+  reverse_pass_callback([inv_m, arena_m, res]() mutable {
+    arena_m.adj().array() -= inv_m * res.adj().array() * res.val().array();
+  });
+  return promote_scalar_t<var, Mat>(res);
+}
+
+///
+
+/**
+ * Return a matrix divided by a matrix elementwise.
+ * @tparam Mat1 Either a type inheriting from `EigenBase` or a `var_value<T>`
+ *  with type `T` inheriting from `EigenBase`.
+ * @tparam Mat2 Either a type inheriting from `EigenBase` or a `var_value<T>`
+ *  with type `T` inheriting from `EigenBase`.
+ * @param[in] m1 specified matrix or expression
+ * @param[in] m2 specified matrix or expression
+ */
+template <typename Mat1, typename Mat2,
+          require_all_matrix_st<is_var_or_arithmetic, Mat1, Mat2>* = nullptr,
+          require_any_matrix_st<is_var, Mat1, Mat2>* = nullptr>
+inline auto divide(const Mat1& m1, const Mat2& m2) {
+  if (!is_constant<Mat1>::value && !is_constant<Mat2>::value) {
```

Comment:
Can you combine these overloads and if statements using templated `arena_t` calls:
```
    arena_t<promote_scalar_t<scalar_type_t<Mat1>, Mat1>> arena_m1 = m1;
```

And using the `if (!is_constant<Mat1>::value)` constant calls with `forward_as`? I'm thinking of how I did the gradients [over here](https://github.com/stan-dev/math/pull/2510/files#diff-d4f2bf898ef6a07300a1aba5e71a511e110449536fcdbace0be0a3afc2a26b8eR34) which dramatically simplified the overloads

Reply:
Plus you can replace calls like:
```
arena_m2.val().array().inverse()
```
With
```
1.0 / as_array_or_scalar(arena_m2.val())
```

To be matrix/scalar agnostic

Reply:
So I like the idea of talking the `(Scalar, Matrix)` and `(Matrix, Scalar)` overloads, but it feels like the functions could become a bit too dense to read if we combine all of them. If we had `if constexpr` I'd be more comfortable doing that. But I did switch these up to put each of the `(Scalar, Matrix)` and `(Matrix, Scalar)` overloads together

Reply:
Yeah I see what you mean